src/ansible_navigator/steps.py

```diff
@@ -122,6 +128,90 @@ def _value_check(value: Any, want: Union[type, Tuple[type, ...]]) -> None:
             raise ValueError(f"wanted {want}, got {type(value)}")
 
 
+TValueType = TypeVar("TValueType")
+
+
+@dataclass
+class TypedStep(Generic[TValueType]):
+    # pylint: disable=too-many-instance-attributes
+    """One step in the flow of things.
+
+    This should be used for new actions following the pattern of a
+    TypedDict for the user interface as seen in settings.
+    """
+
+    name: str
+    step_type: str
```

Comment:
Can this really be any string and still be valid? Or are there only a few valid possibilities it can be? In the latter case, it might be best to use an enum or emulated sum type.

Reply:
Using this now: 
```
class StepType(Enum):
    """Type of step, either menu or content."""

    MENU = "menu"
    CONTENT = "content"
```